h/services/delete_user.py

```diff
@@ -23,17 +23,36 @@ def delete(self, user):
         message.
         """
 
-        if Group.created_by(self.request.db, user).count() > 0:
-            raise UserDeleteError('Cannot delete user who is a group creator.')
-
-        user.groups = []
```

Comment:
This threw me for a second.

I was confused that we were no longer explicitly removing the candidate user’s group memberships, had no explicit configuration for cascading deletes, and yet somehow weren’t seeing integrity errors all over the place as we tried to delete things (as we’ve seen before with `AuthTicket`s and feature flags).

For future reference, I found the answer after a bit of trawling through the SQLAlchemy documentation: deletions [cascade automatically when there’s a `secondary` table configured](http://docs.sqlalchemy.org/en/rel_1_1/orm/basic_relationships.html#deleting-rows-from-the-many-to-many-table). This didn’t apply to the group creator relationship, which was why it was in there originally, and why it’s not needed now. (See the [original commit](ab71a9273444cfcefb588bbfb6e318ae1e4e96f1) for reference.)

That makes a load more sense now, and I understand why we’re not clearing these groups any more.